src/components/course/CourseEnrollmentFailedAlert.jsx

```diff
@@ -1,67 +1,129 @@
 import React, { useContext, useMemo } from 'react';
+import PropTypes from 'prop-types';
 import { useLocation } from 'react-router-dom';
 import { AppContext } from '@edx/frontend-platform/react';
 import { Container, Alert } from '@edx/paragon';
+import { Error } from '@edx/paragon/icons';
 
 import { useRenderContactHelpText } from '../../utils/hooks';
 import {
+  ENROLLMENT_COURSE_RUN_KEY_QUERY_PARAM,
   ENROLLMENT_FAILED_QUERY_PARAM,
   ENROLLMENT_FAILURE_REASON_QUERY_PARAM,
 } from './data/constants';
+import { CourseEnrollmentsContext } from '../dashboard/main-content/course-enrollments/CourseEnrollmentsContextProvider';
+
+export const ENROLLMENT_SOURCE = {
+  DASHBOARD: 'DASHBOARD',
+  COURSE_PAGE: 'COURSE_PAGE',
+};
+
+const createEnrollmentFailureMessages = (contactHelpText) => ({
+  dsc_denied: (
```

Comment:
nit: camel case

Reply:
The query param is snakeCased so keeping this snakecased let's us get away from having to convert